src/systems/menu-animation-system.js

```diff
@@ -0,0 +1,73 @@
+import { waitForDOMContentLoaded } from "../utils/async-utils";
+import { elasticOut } from "../utils/easing";
+const MENU_ANIMATION_DURATION_MS = 750;
+export class MenuAnimationSystem {
+  constructor() {
+    this.components = [];
+    this.data = new Map();
+    this.tick = this.tick.bind(this);
+    waitForDOMContentLoaded().then(() => {
+      this.viewingCamera = document.getElementById("viewing-camera").object3D;
+    });
+  }
+  register(component, menuEl, chooseScale) {
+    this.components.push(component);
+    this.data.set(component, {
+      menuEl,
+      chooseScale,
+      menuOpenTime: -1,
+      startScaleAtMenuOpenTime: 0,
+      wasMenuVisible: false,
+      endingScale: 0
+    });
+  }
+  unregister(component) {
+    this.components.splice(this.components.indexOf(component), 1);
+    this.data.delete(component);
+  }
+  tick = (function() {
+    const menuToCamera = new THREE.Vector3();
+    const menuScale = new THREE.Vector3();
+    const menuParentScale = new THREE.Vector3();
+    const menuPosition = new THREE.Vector3();
+    const cameraPosition = new THREE.Vector3();
+    return function tick(t) {
+      if (!this.viewingCamera) {
+        return;
+      }
+      this.viewingCamera.updateMatrices();
+      cameraPosition.setFromMatrixPosition(this.viewingCamera.matrixWorld);
+
+      for (let i = 0; i < this.components.length; i++) {
+        const datum = this.data.get(this.components[i]);
```

Comment:
Seems like the only time we ever get the component data is here when iterating, so we can probably ditch the array + Map and just use an array with the data. You will need to iterate to find the item again for `unregister` but you are already calling `indexOf` on the array anyway, so it already is.